### ISSUE TITLE
Flathub commit 5c6ca46 broke Fluffychat: libjsoncpp.so.25: cannot open shared object file

### ISSUE THREAD

**Issue by mbuesch** [flathub/im.fluffychat.Fluffychat#60]
The flathub commit f`5c6ca46a8f6400db6e90f69a606fff90cd45960f5f308be0c4a051de21dd5bf6` broke Fluffychat:

```
$ flatpak run im.fluffychat.Fluffychat 
fluffychat: error while loading shared libraries: libjsoncpp.so.25: cannot open shared object file: No such file or directory
```

Downgrading to the previous commit fixes it:

```
flatpak update --commit=9e8c890b9e75438d0a1c6f160093a4c130d2e0c2d3c174d166a59c1c316f83f5 im.fluffychat.Fluffychat
```

**Comment by Mikaela** [comment]
Workaround: https://github.com/flathub/im.fluffychat.Fluffychat/pull/59 requiring flathub-beta

**Comment by mbuesch** [comment]
Well, I'm sorry. That's not an actual workaround.

Fluffychat is currently broken by a recent commit. This is not a commit that broke some random small detail of the package. The commit made it *completely useless*. People can't use Fluffychat anymore. Normal (non-IT) people won't know how to downgrade.
And you really cannot expect people installing even more beta stuff to fix this.

The only correct reaction in my opinion is to revert that commit.
Everybody does mistakes. Including me. But it should be priority number 1 then to revert to a previously working state.

I'm very sorry for my grumpy attitude. I really don't mean to attack you personally. But this happens a lot quite lately with many software projects. This is not how high quality software should be developed.

**Comment by Rho-n** [comment]
> The flathub commit f`5c6ca46a8f6400db6e90f69a606fff90cd45960f5f308be0c4a051de21dd5bf6` broke Fluffychat:
> 
> ```
> $ flatpak run im.fluffychat.Fluffychat 
> fluffychat: error while loading shared libraries: libjsoncpp.so.25: cannot open shared object file: No such file or directory
> ```

I am seeing this same error. Currently using flatpak on Pop!_OS 22.04 LTS
 
> Downgrading to the previous commit fixes it:
> 
> ```
> flatpak update --commit=9e8c890b9e75438d0a1c6f160093a4c130d2e0c2d3c174d166a59c1c316f83f5 im.fluffychat.Fluffychat
> ```

I can confirm using this commit did resolve the issue as far as getting Fluffychat running again. Thanks!

**Comment by Caticer** [comment]
I have same issue

**Comment by mbuesch** [comment]
Is this project even maintained?
An automated bot @flathubbot broke it and nobody cares.

Please remove this package from flathub, if it is not maintained.

**Comment by MTRNord** [comment]
> Is this project even maintained? An automated bot @flathubbot broke it and nobody cares.
> 
> Please remove this package from flathub, if it is not maintained.

It is maintained very actively. We are just a) at gitlab with our sourcecode and b) if there are no issues, there is no need for manual actions over here.

Also: Your comment is incredibly rude.... You dont seem to be contributing to a healthy community to say the least.

And if you want to see that we are active feel free to checkout this: https://gitlab.com/famedly/fluffychat/-/commits/main

**Comment by mbuesch** [comment]
> It is maintained very actively.

Ok, That's very good. Thank you very much for your work.

> We are just a) at gitlab with our sourcecode and 

Ok. Interesting.
The link from flathub.org points to github and here's no mention that this is not actually the right place.

https://flathub.org/apps/details/im.fluffychat.Fluffychat

Could you please fix the link on flathub.org, so that it points your users to the correct location? Thanks.

> b) if there are no issues, there is no need for manual actions over here.

Well, there obviously is something to do over here.

> Also: Your comment is incredibly rude

You completely broke the software that people depend on and you don't do anything about that.
*That's* very rude, indeed.

This will drive people away from FluffyChat.

I like FluffyChat very much. But over one week ago it broke on flathub. And today it broke on Android, too!
FluffyChat is currently completely unusable. It can't get any worse.

Therefore, all my setups are currently broken on all of my machines. I'm certainly not the only one with these problems.

> You dont seem to be contributing to a healthy community to say the least.

A healthy community with a completely broken piece of software that doesn't work at all since over a week.
Doesn't sound very healthy to me. I'm sorry. I am not the problem here. I am the user.

> And if you want to see that we are active feel free to checkout this

Please revert the breaking change. It's that simple. That should have happened 7 days ago already.

It's not Ok to leave a package in a broken state for so long, if a simple revert can temporarily fix it.

**Comment by MTRNord** [comment]
> I like FluffyChat very much. But over one week ago it broke on flathub. And today it broke on Android, too!

This happens in Software. We are an incredibly small team and this is a hobby project being done. We do this for fun and not for full time. Don't expect fixes in a day please. We have to do work while maintaining this and have a life. We don't get payed for this either.

> I am not the problem here. I am the user.

Your tone sets the mood here. And I am honest: if I were the core maintainer of this (which I am not as I did turn down as a developer of fluffy a whole back due to people like you) I would not want you as a user anyway.

> Please revert the breaking change. It's that simple.

It's not that simple.

**Comment by mbuesch** [comment]
> This happens in Software

Bugs happen. Yes.
But not reverting them doesn't simply *happen*. It's a conscious decision.
You decided not to revert the publicly visible breaking change, but instead work on fixing the bug in the development git repository. Your users are left with a completely broken setup in the mean time.
That doesn't just *happen*.

> Don't expect fixes in a day please.

I don't and I never did.
But I expect a revert of a breaking change within reasonable time (= ASAP).

> would not want you as a user anyway.

That could be interpreted as being rude, you know?

> It's not that simple.

Yes, it is.
In the initial post of this issue I even wrote the hash of the last commit that worked.
How hard is it to instruct flathub to deliver that one by default, instead of the latest one?
That would fix the user facing problem.

To make my point clear again:
I do *not* expect a bug fix from you any time soon. Please take all the time it needs to properly fix the bug. That's all up to you and I'm fine with that.
But I *do* expect a revert of the breaking change ASAP.

Everything else is just plain rude to your users.

I am also an Open Source software developer since over two decades. Therefore, I know how easy it can sometimes be to break things. But it must absolutely be the number 1 priority to revert breaking changes. This is not the 80s and 90s anymore, where nobody cared and nobody used Open Source software for anything serious.
This is 2022, where people use this software for things they depend on.

If I release a new version of any of the software I maintain and it breaks UX in a serious way (i.e. completely broken, not usable, etc...), then I will immediately pull it from the distribution channels. And I think that's the only correct reaction to it. Hobby Open Source developers *do* have a responsibility. It is not correct to say "I'm doing this only for a hobby, therefore it's ok leave everything broken as long as I desire". If something broke by a change, it's mandatory to either fix it or revert it ASAP.

If you don't change your policy to revert breaking changes, then I will have to migrate away from FluffyChat. I don't want to, because I think it really is a nice software. But this would force me to do so.

I'm probably not the only person with that opinion.

**Comment by mbuesch** [comment]
@Raatty + @kate-shine
Thank you very much for reverting the flathub package back to 1.5.0.
The default package install works again. 
:+1:

**Comment by mbuesch** [comment]
I'll close this issue, as the problem from the original post has been solved.
If you want to keep this as a tracking issue for the fix in 1.6.0, please re-open.
